### PR TITLE
Estimate min/max value for numeric exposes with set access

### PR DESCRIPTION
**Pull request by Koenkk**
Attempt to provide a good estimate of the min/max value for numeric exposes which are settable. This should prevent HA complaining about the values being out of the default range (0-100) and improve the slider experience in the frontend/HA.

Also added a test case to ensure all settable numerics have a min/max value.

Closes https://github.com/Koenkk/zigbee2mqtt/issues/9972

CC: @sjorge @robertalexa @nurikk

### CODE REVIEW

File: devices/danfoss.js [hunk 1]
```diff
@@ -45,7 +45,8 @@ module.exports = [
                 .withSystemMode(['heat']).withRunningState(['idle', 'heat'], ea.STATE),
             exposes.numeric('external_measured_room_sensor', ea.ALL)
                 .withDescription('Set at maximum 3 hours interval but not more often than every 30 minutes and 0.1 ' +
-                    'degrees difference. Resets every 3hours to standard. e.g. 21C = 2100 (-8000=undefined).'),
+                    'degrees difference. Resets every 3hours to standard. e.g. 21C = 2100 (-8000=undefined).')
+                .withValueMin(-8000).withValueMax(35),
```
Comment (sjorge) [inline devices/danfoss.js:49]: Max should be `3500` I think? Given that 21C = 2100 according to the description.

Comment (sjorge) [reply devices/danfoss.js:49]: Same for other thermostats based on this design

File: devices/immax.js [hunk 2]
```diff
@@ -148,10 +148,14 @@ module.exports = [
         fromZigbee: [fz.battery, fz.ignore_basic_report, fz.illuminance, fz.ZB003X, fz.ZB003X_attr, fz.ZB003X_occupancy],
         toZigbee: [tz.ZB003X],
         exposes: [e.occupancy(), e.tamper(), e.battery(), e.illuminance(), e.illuminance_lux().withUnit('lx'), e.temperature(),
-            e.humidity(), exposes.numeric('reporting_time', ea.STATE_SET).withDescription('Reporting interval in minutes'),
-            exposes.numeric('temperature_calibration', ea.STATE_SET).withDescription('Temperature calibration'),
-            exposes.numeric('humidity_calibration', ea.STATE_SET).withDescription('Humidity calibration'),
-            exposes.numeric('illuminance_calibration', ea.STATE_SET).withDescription('Illuminance calibration'),
+            e.humidity(), exposes.numeric('reporting_time', ea.STATE_SET).withDescription('Reporting interval in minutes')
+                .withValueMin(0).withValueMax(1440),
+            exposes.numeric('temperature_calibration', ea.STATE_SET).withDescription('Temperature calibration')
```
Comment (sjorge) [inline devices/immax.js:153]: I'm confused, is this a internal device thing? Or should these really not be expose data but options ?

File: devices/danfoss.js (hunk 1 above)
Comment (robertalexa) [reply devices/danfoss.js:49]: Should be -8000 to 3500.

Same for Hive and Popp

File: test/index.test.js [hunk 3]
```diff
@@ -452,4 +452,19 @@ describe('index.js', () => {
         const definition = index.findByZigbeeModel('MCT-340 SMA');
         expect(index.getConfigureKey(definition)).toBe(1);
     });
+
+    it('Number exposes with set access should have a range', () => {
+        index.definitions.forEach((device) => {
+            if (device.exposes) {
+                const toCheck = [];
```
Comment (nurikk) [inline test/index.test.js:459]: Redundant variable

**Comment by robertalexa** [comment]
I have left a few comments, apart from that, it looks good (as far as i can tell considering I am not familiar with some of the devices involved)

> This should prevent HA complaining

Currently because of those limits some of the attributes become unusable completely, so not just a warning. Based on that, it might be worth doing a hotfix release rather than wait for the usual 1st of the month?

Thanks for looking into this Koen

File: devices/immax.js (hunk 2 above)
Comment (Koenkk) [reply devices/immax.js:153]: this device support calibration on the device side (instead of on the z2m side as usual)

**Comment by Koenkk** [comment]
> Currently because of those limits some of the attributes become unusable completely, so not just a warning. Based on that, it might be worth doing a hotfix release rather than wait for the usual 1st of the month?

I only want to do this when many users are affected, I believe this is not the case. Those who are affected can use the dev branch for the time being.

**Comment by robertalexa** [comment]
@Koenkk There was another review I have started which you have not answered

image

**Comment by Koenkk** [comment]
@robertalexa I think you need to click the submit review button, comments that are Pending are not visible to others.

**Comment by robertalexa** [comment]
Oh cr*p. Didn't realise that pending review are not visible. Please accept my apologies

File: devices/danfoss.js [hunk 4]
```diff
@@ -64,7 +65,8 @@ module.exports = [
                 .withDescription('Whether or not the thermostat acts as standalone thermostat or shares load with other ' +
                     'thermostats in the room. The gateway must update load_room_mean if enabled.'),
             exposes.numeric('load_room_mean', ea.ALL)
-                .withDescription('Mean radiator load for room calculated by gateway for load balancing purposes'),
+                .withDescription('Mean radiator load for room calculated by gateway for load balancing purposes')
+                .withValueMin(-30).withValueMax(30),
```
[the remark this answers is not in the document]
Comment (Koenkk) [reply devices/danfoss.js:69]: Done, thanks!

**Pull request merged**